### PR TITLE
Merging Auth Branch with everything

### PR DESCRIPTION
**Pull request by KeithSantamaria**

### CODE REVIEW

**Review by encperez — APPROVED** [review]
Technically everything but devops

File: backend/.idea/backend.iml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="JAVA_MODULE" version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="true">
+    <exclude-output />
+    <content url="file://$MODULE_DIR$" />
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
```
Comment (KAltieri) [inline backend/.idea/backend.iml:7]: based

**Review by tomjmartinez — APPROVED** [review]

**Review by KeithSantamaria — COMMENTED** [review]
Just removed the lists on the model- GJ!

File: backend/UserService/src/main/java/com/example/userservice/controller/UserController.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import lombok.extern.log4j.Log4j2;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@Log4j2
+@RestController
+@RequestMapping("/user")
+//CHANGE THAT ONCE WE GET FRONTEND STABLE
+@CrossOrigin(origins = "http://localhost:3000", maxAge = 3600)
```
Comment (KAltieri) [inline backend/UserService/src/main/java/com/example/userservice/controller/UserController.java:14]: Revisit this soon

File: backend/UserService/src/main/resources/application.properties [hunk 3]
```diff
@@ -1 +1,4 @@
 server.port=6969
```
Comment (KAltieri) [inline backend/UserService/src/main/resources/application.properties:1]: Sticking with local mongo or switching to atlas?

File: frontend/src/components/ERDCard.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import { setERDiagrams } from "../redux/projectAppSlice";
+
+const ERDCard = ({ erd }: any) => {
+  const dispatch = useAppDispatch();
+  const [erdName] = React.useState(erd.erdName);
+  const [erdImg] = React.useState(erd.erdImageUrl);
+  const [erdDescription] = React.useState(erd.erdDescription);
+
+  const removeERD = (id: String) => {
+    const queryString = `http://localhost:42069/api/delete/project/ERD`;
```
Comment (KAltieri) [inline frontend/src/components/ERDCard.tsx:14]: Update

File: frontend/src/components/ERDForm.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+  const formChangeHandler = (event: any) => {
+    const fieldName = event.target.name;
+    const value = event.target.value;
+    dispatch(setCreateNewERDForm({ fieldName, value }));
+  };
+
+  const addERD = () => {
+    // Test
+    const projectId = "60bc36b65d2b0da1deb9ada2";
```
Comment (KAltieri) [inline frontend/src/components/ERDForm.tsx:25]: Needs to be updated soon - update to the user's project's id

File: frontend/src/components/EndpointForm.tsx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+  const formChangeHandler = (event: any) => {
+    const fieldName = event.target.name;
+    const value = event.target.value;
+    dispatch(setCreateNewEndpointForm({ fieldName, value }));
+  };
+
+  const addEndpoint = () => {
+    // Test
+    const projectId = "60bc36b65d2b0da1deb9ada2";
```
Comment (KAltieri) [inline frontend/src/components/EndpointForm.tsx:24]: Make this dynamic to the user / project that is being worked upon

File: frontend/src/components/ModelForm.tsx [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+  const formChangeHandler = (event: any) => {
+    const fieldName = event.target.name;
+    const value = event.target.value;
+    dispatch(setCreateNewModelForm({ fieldName, value }));
+  };
+
+  const addModel = () => {
+    // Test
+    const projectId = "60bc36b65d2b0da1deb9ada2";
```
Comment (KAltieri) [inline frontend/src/components/ModelForm.tsx:24]: Update

File: frontend/src/components/ProjectERDs.tsx [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+const ProjectERDs = () => {
+  const projectAppState = useAppSelector(selectProjectApp);
+  const [modalShow, setModalShow] = React.useState(false);
+  const handleOpen = () => setModalShow(true);
+  const handleClose = () => setModalShow(false);
+  const dispatch = useAppDispatch();
+
+  const getERDs = () => {
+    //Test
+    const projectId = "60bc36b65d2b0da1deb9ada2";
```
Comment (KAltieri) [inline frontend/src/components/ProjectERDs.tsx:19]: Update

File: frontend/src/components/ProjectERDs.tsx [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+        onHide={handleClose}
+      >
+        <Modal.Header closeButton>
+          <Modal.Title id="contained-modal-title-vcenter">
+            Create New ERD
+          </Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          {/* Production */}
+          {/* <EndpointForm projectId={projectAppState.project.projectId} /> */}
```
Comment (KAltieri) [inline frontend/src/components/ProjectERDs.tsx:62]: Review

File: frontend/src/components/ProjectEndpoints.tsx [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+  const handleOpen = () => setModalShow(true);
+  const handleClose = () => setModalShow(false);
+  const dispatch = useAppDispatch();
+
+  const getEndpoints = () => {
+    // Test query string works; comment when ready to test prod
+    const queryString = `http://localhost:42069/api/read/project/endpoints`;
+    const body = {
+      params: {
+        projectId: "60bc36b65d2b0da1deb9ada2",
```
Comment (KAltieri) [inline frontend/src/components/ProjectEndpoints.tsx:22]: Update

File: frontend/src/components/ProjectGeneralInfo.tsx [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+  const handleOpen = () => setModalShow(true);
+  const handleClose = () => setModalShow(false);
+
+  const getProject = () => {
+    // Production
+    // const queryString = `http://localhost:42069/api/read/project?projectId=${projectId}`;
+
+    // Test
+    const queryString = `http://localhost:42069/api/read/project`;
+    const projectId = "60bc36b65d2b0da1deb9ada2";
```
Comment (KAltieri) [inline frontend/src/components/ProjectGeneralInfo.tsx:32]: Update

File: frontend/src/components/ProjectModels.tsx [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+  const handleClose = () => setModalShow(false);
+  const dispatch = useAppDispatch();
+
+  const getModels = () => {
+    // Production
+    // const queryString = `http://localhost:42069/api/read/project?projectId=${projectId}`;
+
+    // Test
+    const queryString = `http://localhost:42069/api/read/models`;
+    const projectId = "60bc36b65d2b0da1deb9ada2";
```
Comment (KAltieri) [inline frontend/src/components/ProjectModels.tsx:23]: Update

File: frontend/src/components/ProjectUserStories.tsx [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+  const [modalShow, setModalShow] = React.useState(false);
+  const handleOpen = () => setModalShow(true);
+  const handleClose = () => setModalShow(false);
+  const dispatch = useAppDispatch();
+
+  const getUserStories = () => {
+    const queryString = `http://localhost:42069/api/read/project/userstory`;
+    const body = {
+      params: {
+        projectId: "60bc36b65d2b0da1deb9ada2",
```
Comment (KAltieri) [inline frontend/src/components/ProjectUserStories.tsx:21]: Update

File: frontend/src/components/ProjectWireframes.tsx [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+function ProjectWireframes() {
+  const projectAppState = useAppSelector(selectProjectApp);
+  const [modalShow, setModalShow] = React.useState(false);
+  const handleOpen = () => setModalShow(true);
+  const handleClose = () => setModalShow(false);
+  const dispatch = useAppDispatch();
+
+  const getWireframes = () => {
+    //Test
+    const projectId = "60bc36b65d2b0da1deb9ada2";
```
Comment (KAltieri) [inline frontend/src/components/ProjectWireframes.tsx:19]: Update

File: frontend/src/components/ProjectsHome.tsx [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+import ProjectUserStories from './ProjectUserStories';
+import ProjectTags from './ProjectTags';
+import ProjectEndpoints from './ProjectEndpoints';
+
+function ProjectsHome() {
+    return (
+        <div>           
+            <Router>
+                <Switch>
+                    {/*This will change once Project Services is hooked up to home*/}
```
Comment (KAltieri) [inline frontend/src/components/ProjectsHome.tsx:17]: ?

File: frontend/src/components/TagForm.tsx [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+
+    const formChangeHandler = (event: any) => {
+        const fieldName = event.target.name;
+        const value = event.target.value;
+        dispatch(setCreateNewTagForm({ fieldName, value }));
+    };
+
+    const addTag = () => {
+        //Test
+        const projectId = "60bc36b65d2b0da1deb9ada2";
```
Comment (KAltieri) [inline frontend/src/components/TagForm.tsx:24]: Update

File: frontend/src/components/UserStoryForm.tsx [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+
+  const formChangeHandler = (event: any) => {
+    const fieldName = event.target.name;
+    const value = event.target.value;
+    dispatch(setCreateNewUserStoryForm({ fieldName, value }));
+  };
+
+  const addUserStory = () => {
+    // Test
+    const projectId = "60bc36b65d2b0da1deb9ada2";
```
Comment (KAltieri) [inline frontend/src/components/UserStoryForm.tsx:25]: Update

File: frontend/src/components/WidgetComponent.tsx [hunk 18]
```diff
@@ -0,0 +1,55 @@
...
+import { Card } from 'react-bootstrap';
+
+const WidgetComponent = ({projectId}: any) => {
+    const dispatch = useAppDispatch();
+    const projectAppState= useAppSelector(selectProjectApp);
+
+    const getWidgets = () => {
+        // Test query string works; comment when ready to test prod
+        const queryString = `http://localhost:42069/api/read/project/widgets`;
+        const projectId = "60bc36b65d2b0da1deb9ada2";
```
Comment (KAltieri) [inline frontend/src/components/WidgetComponent.tsx:14]: Update

File: frontend/src/components/WidgetForm.tsx [hunk 19]
```diff
@@ -0,0 +1,79 @@
...
+    if (
+      projectAppState.createNewWidgetForm.widgetDescription === "" ||
+      projectAppState.createNewWidgetForm.widgetName === ""
+    ) {
+      alert("There is nothing to add");
+    } else {
+      const widget = {
+        widgetName: projectAppState.createNewWidgetForm.widgetName,
+        widgetDescription: projectAppState.createNewWidgetForm.widgetDescription,
+        projectId: "60bc36b65d2b0da1deb9ada2"
```
Comment (KAltieri) [inline frontend/src/components/WidgetForm.tsx:34]: Update

File: frontend/src/components/WireframeForm.tsx [hunk 20]
```diff
@@ -0,0 +1,121 @@
...
+  const formChangeHandler = (event: any) => {
+    const fieldName = event.target.name;
+    const value = event.target.value;
+    dispatch(setCreateNewWireframeForm({ fieldName, value }));
+  };
+
+  const addWireframe = () => {
+    // Test
+    const projectId = "60bc36b65d2b0da1deb9ada2";
+    const queryString = `http://localhost:42069/api/create/project/wireframe`;
```
Comment (KAltieri) [inline frontend/src/components/WireframeForm.tsx:26]: Update

**Review by KAltieri — APPROVED** [review]
Based

**Review by cowelch1 — APPROVED** [review]

**Review by Basic-Sombrero — APPROVED** [review]
yee

**Pull request merged**